src/FFI.php

```diff
@@ -169,6 +178,30 @@ public static function atLeast(int $x, int $y, int $z = 0): bool
                 self::$library_micro >= $z);
     }
 
+    /**
+     * Adds a directory to the search path for shared libraries.
+     *
+     * This method has no effect if FFI handles are already initialized or
+     * if the specified path is already included.
+     *
+     * @param string $path The path of the library.
+     * @return bool `true` if the path was added; otherwise, `false`.
+     */
+    public static function addLibraryPath(string $path): bool
+    {
+        // Already initialized.
+        if (self::$ffi_inited) {
+            return false;
+        }
+
```

Comment:
How about validating `$path` before adding it? Eg. 

- it must be an absolute path
- it must not be `""`
- it must not end in `/`

if we have some guarantees about the path, we can increase safety and save some checks later.

Reply:
Good idea! Done with commit deaf4e15d030a8ca923b1e89681d74861a1257a8. It uses `realpath()` for all these checks, which also works with empty strings, e.g. calling `realpath("");` would return the current directory.